### PR TITLE
Add more missing Gtk types

### PR DESCRIPTION
**Pull request by Susurrus**
Partially address #442.

Haven't used any of this code, but didn't see any warnings about generating it. There were some GTK types that can't be generated, so I commented them out. Should I add them to the `ignore` list instead?

### CODE REVIEW

File: Gir.toml [hunk 1]
```diff
@@ -175,7 +182,13 @@ generate = [
     "Gtk.StateType",
     "Gtk.Statusbar",
     "Gtk.StatusIcon",
+    "Gtk.Style",
```
Comment (EPashkin) [inline Gir.toml]: Its deprecated in 3.0, so better don't generate.

File: src/auto/border.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// This file was generated by gir (8a0b5e5) from gir-files (71d73f0)
+// DO NOT EDIT
+
+use ffi;
+use glib::translate::*;
+
+glib_wrapper! {
+    pub struct Border(Boxed<ffi::GtkBorder>);
```
Comment (EPashkin) [inline src/auto/border.rs]: Without fields it IMHO unusable

File: src/auto/target_list.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    pub struct TargetList(Shared<ffi::GtkTargetList>);
+
+    match fn {
+        ref => |ptr| ffi::gtk_target_list_ref(ptr),
+        unref => |ptr| ffi::gtk_target_list_unref(ptr),
+    }
+}
+
+impl TargetList {
+    //pub fn new(targets: /*Ignored*/&[&TargetEntry], ntargets: u32) -> TargetList {
```
Comment (EPashkin) [inline src/auto/target_list.rs]: No reason to generate record without constructors

**Comment by EPashkin** [comment]
Strange that it builds, as I remember `WidgetPath` has mixed `const/mut*` usage and fails.

**Comment by Susurrus** [comment]
I think I resolved all of your comments, so here's another batch of changes. Sorry about not checking over the generated code, I looked over the remaining generated types and they all seemed to have enough useful methods to be worthwhile.

**Comment by GuillaumeGomez** [comment]
Seems ok for me. Anything else for you @EPashkin?

**Comment by EPashkin** [comment]
style_context.rs don't builds on gtk/v3_20.
Function `to_string` conflict in `impl fmt::Display for StyleContext`
`Gir` need change to don't generate this impl for nonstandard functions.
@GuillaumeGomez, seems all good.

**Comment by GuillaumeGomez** [comment]
@EPashkin: I merge then.
@Susurrus: Thanks!

**Comment by Susurrus** [comment]
Given the fix in #313 I think there'll need to be another regent to make StylrContext correct, yes?

**Comment by EPashkin** [comment]
I already made [PR](https://github.com/gtk-rs/gtk/pull/444), but forgot link this.

**Pull request merged**